### PR TITLE
add filter for binary_linux_wheel_py3.7_cpu

### PR DESCRIPTION
**Pull request by mattip**
fixes issue #3997

The important part is to add a filter for tags, otherwise the tag commit does not trigger a documentation build and upload.

### CODE REVIEW

File: .circleci/regenerate.py [hunk 1]
```diff
@@ -45,6 +45,10 @@ def build_workflows(prefix='', filter_branch=None, upload=False, indentation=6,
                             (python_version != python_versions[-1] or
                              (cu_version not in [cu_versions[0], cu_versions[-1]])):
                             fb = "master"
+                        if not fb and (os_type == 'linux' and cu_version == 'cpu' and
+                            btype == 'wheel' and python_version =='3.7'):
+                            # this is for build_docs, the versions must match
```
Comment (NicolasHug) [inline .circleci/regenerate.py:50]: I'm not sure I understand this comment: what is "versions", and what should it match with?

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @mattip!

I'm trying to understand a few things:

> The important part is to add a filter for tags, otherwise the tag commit does not trigger a documentation build and upload.

Why do we only need the tag for `binary_linux_wheel_py3.7_cpu`?

The [CircleCI docs](https://circleci.com/docs/2.0/configuration-reference/#jobfilters) say:

> CircleCI does not run workflows for tags unless you explicitly specify tag filters. Additionally, if a job requires any other jobs (directly or indirectly), you must specify tag filters for those jobs.

So, since the dependency between our jobs is upload_docs -> build_docs -> binary_linux_wheel_py3.7_cpu, shouldn't we also add the tag filter for build_docs, which is currently missing?

File: .circleci/regenerate.py (hunk 1 above)
Comment (mattip) [inline .circleci/regenerate.py:50]: Is this clearer?
```suggestion
                            # the fields must match the build_docs "requires" dependency
```

**Comment by mattip** [comment]
> shouldn't we also add the tag filter for build_docs, which is currently missing?

Good point. Added and rebased off master.

**Review by datumbox — COMMENTED** [review]
@mattip Thanks for fixing it. There are some related [lint issues](https://app.circleci.com/pipelines/github/pytorch/vision/8778/workflows/8b1f1740-69e5-4c9f-a592-20a2a602072b/jobs/637200). Could you please check?

**Review by datumbox — APPROVED** [review]
LGTM, thanks!

**Comment by seemethere** [comment]
So the way we actually do this in upstream `pytorch/pytorch` is by specifying a `/.*/` branch in the filter branch to run on all branches:

https://github.com/pytorch/pytorch/blob/b844fd11eeacf0391fd904e5508bda1af54664f8/.circleci/config.yml#L6780-L6787

File: .circleci/config.yml [hunk 2]
```diff
@@ -1015,6 +1015,11 @@ workflows:
       - binary_linux_wheel:
           conda_docker_image: pytorch/conda-builder:cpu
           cu_version: cpu
+          filters:
+            branches:
+              only: master,nightly
```
Comment (malfet) [inline .circleci/config.yml:1020]: Can we add `only: *` here to keep this job available on all branches but also make it an opt-in for tags

Comment (mattip) [reply .circleci/config.yml:1020]: done

**Comment by mattip** [comment]
Anything else needed here? It seems this goes out-of-date every night, so if needed I can update

**Review by malfet — APPROVED** [review]

File: .circleci/regenerate.py [hunk 3]
```diff
@@ -45,13 +45,19 @@ def build_workflows(prefix='', filter_branch=None, upload=False, indentation=6,
...
+                            # the fields must match the build_docs "requires" dependency
+                            fb = "*"
                         w += workflow_pair(
                             btype, os_type, python_version, cu_version,
                             unicode, prefix, upload, filter_branch=fb)
 
     if not filter_branch:
         # Build on every pull request, but upload only on nightly and tags
-        w += build_doc_job(None)
+        w += build_doc_job('nightly')
```
Comment (fmassa) [inline .circleci/regenerate.py:60]: @mattip does this mean that the docs are not build on every PR anymore?

Comment (mattip) [reply .circleci/regenerate.py:60]: hmm. It seems to be building on PRs, for instance taking a random PR and looking at the [CI build](https://app.circleci.com/pipelines/github/pytorch/vision/8948/workflows/e9471ac9-00fd-4c07-993b-dbff08add903) I see a `build_docs` run 2/3 of the way down. Although I agree the filter says only nightly, so it is misleading.

```
      - build_docs:
          filters:
            branches:
              only:
              - nightly
            tags:
              only: /v[0-9]+(\.[0-9]+)*-rc[0-9]+/
          name: build_docs
          python_version: '3.7'
          requires:
          - binary_linux_wheel_py3.7_cpu

```

Comment (NicolasHug) [reply .circleci/regenerate.py:60]: I'm afraid the PR you're linking to was pushed before this one got merged.

In more recent commits, e.g. https://app.circleci.com/pipelines/github/pytorch/vision?branch=pull%2F3808 from this morning, I do not see the `build_docs` job.

**Pull request merged**